MiBench/automotive/basicmath/src/basicmath_large.c

```diff
@@ -90,34 +99,40 @@ int main(void)
 
 
   printf("********* INTEGER SQR ROOTS ***********\n");
-  /* perform some integer square roots */
-  for (i = 0; i < 100000; i+=2)
-    {
-      usqrt(i, &q);
-			// remainder differs on some machines
-     // printf("sqrt(%3d) = %2d, remainder = %2d\n",
-     printf("sqrt(%3d) = %2d\n",
-	     i, q.sqrt);
-    }
+    /* perform some integer square roots */
+  start = clock();
+  for (i = 0; i < 100000; i += 2)
+  {
+    usqrt(i, &q);
+    // remainder differs on some machines
+      // printf("sqrt(%3d) = %2d, remainder = %2d\n",
+    printf("sqrt(%3d) = %2d\n", i, q.sqrt);
+  }
   printf("\n");
   for (l = 0x3fed0169L; l < 0x3fed4169L; l++)
     {
 	 usqrt(l, &q);
 	 //printf("\nsqrt(%lX) = %X, remainder = %X\n", l, q.sqrt, q.frac);
 	 printf("sqrt(%lX) = %X\n", l, q.sqrt);
     }
-
-
+    end = clock();
+    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
+    total_cpu_time += cpu_time_used;
   printf("********* ANGLE CONVERSION ***********\n");
-  /* convert some rads to degrees */
+   /* convert some rads to degrees */
 /*   for (X = 0.0; X <= 360.0; X += 1.0) */
+  start = clock();
   for (X = 0.0; X <= 360.0; X += .001)
     printf("%3.0f degrees = %.12f radians\n", X, deg2rad(X));
   puts("");
-/*   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 180)) */
+  /*   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 180)) */
   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 5760))
     printf("%.12f radians = %3.0f degrees\n", X, rad2deg(X));
-  
-  
+  end = clock();
+  end_main=clock(); //timestamp to end main
+  cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
+  total_cpu_time += cpu_time_used; 
+  printf("Time taken for startup: %2f seconds\n", cpu_time_used_main);
```

Comment:
This printout does not make sense. You want to measure the time from when the runtime starts loading the wasm file until it starts execute. This means you need to instrument each runtime to print this timestamp and then you need to calaculate the time afterwards.